### PR TITLE
Update class mapping example (modelling the class hierarchy)

### PR DESCRIPTION
**Pull request by jankornblum**
Updated code example for Extbase class / table mapping rrelated to #87623 (Replace config.persistence.classes typoscript configuration).

### CODE REVIEW

File: Documentation/6-Persistence/5-modeling-the-class-hierarchy.rst [hunk 1]
```diff
@@ -82,41 +82,36 @@ The name of the field that contains the type can be chosen freely. In our case,
...
-        }
-        MyVendor\MyExtension\Domain\Model\ScientificInstitution {
-            mapping {
-                tableName = tx_myextension_domain_model_party
-                recordType = \MyVendor\MyExtension\Domain\Model\ScientificInstitution
-            }
-        }
-    }
+You have to tell Extbase for every concrete class in which table the
+data of the instances are stored and with which type they should be stored. This is done in "Configuration/Extbase/Persistence/Classes.php".
```
Comment (mbrodala) [inline Documentation/6-Persistence/5-modeling-the-class-hierarchy.rst:86]:
```suggestion
data of the instances are stored and with which type they should be stored. This is done in :file:`Configuration/Extbase/Persistence/Classes.php`.
```

File: Documentation/6-Persistence/5-modeling-the-class-hierarchy.rst [hunk 2]
```diff
@@ -82,41 +82,36 @@ The name of the field that contains the type can be chosen freely. In our case,
...
+    <?php
+    declare(strict_types = 1);
+
+    return [
+        \MyVendor\MyExtension\Domain\Model\Organization::class => [
+            'tableName' => 'tx_myextension_domain_model_party',
+            'recordType' => '\MyVendor\MyExtension\Domain\Model\Organization',
+            'subclasses' => [
+                '\MyVendor\MyExtension\Domain\Model\Company' => \MyVendor\MyExtension\Domain\Model\Company::class,
+                '\MyVendor\MyExtension\Domain\Model\ScientificInstitution' => MyVendor\MyExtension\Domain\Model\ScientificInstitution::class
```
Comment (mbrodala) [inline Documentation/6-Persistence/5-modeling-the-class-hierarchy.rst:99]:
```suggestion
                '\MyVendor\MyExtension\Domain\Model\ScientificInstitution' => MyVendor\MyExtension\Domain\Model\ScientificInstitution::class,
```

File: Documentation/6-Persistence/5-modeling-the-class-hierarchy.rst [hunk 3]
```diff
@@ -82,41 +82,36 @@ The name of the field that contains the type can be chosen freely. In our case,
...
+            ]
+        ],
+        \MyVendor\MyExtension\Domain\Model\Person::class => [
+            'tableName' => 'tx_myextension_domain_model_party',
+            'recordType' => '\MyVendor\MyExtension\Domain\Model\Person'
```
Comment (mbrodala) [inline Documentation/6-Persistence/5-modeling-the-class-hierarchy.rst:104]:
```suggestion
            'recordType' => '\MyVendor\MyExtension\Domain\Model\Person',
```

File: Documentation/6-Persistence/5-modeling-the-class-hierarchy.rst [hunk 4]
```diff
@@ -82,41 +82,36 @@ The name of the field that contains the type can be chosen freely. In our case,
...
+        ],
+        \MyVendor\MyExtension\Domain\Model\Company::class => [
+            'tableName' => 'tx_myextension_domain_model_party',
+            'recordType' => '\MyVendor\MyExtension\Domain\Model\Company'
```
Comment (mbrodala) [inline Documentation/6-Persistence/5-modeling-the-class-hierarchy.rst:108]:
```suggestion
            'recordType' => '\MyVendor\MyExtension\Domain\Model\Company',
```

File: Documentation/6-Persistence/5-modeling-the-class-hierarchy.rst [hunk 5]
```diff
@@ -82,41 +82,36 @@ The name of the field that contains the type can be chosen freely. In our case,
...
+        ],
+        \MyVendor\MyExtension\Domain\Model\ScientificInstitution::class => [
+            'tableName' => 'tx_myextension_domain_model_party',
+            'recordType' => '\MyVendor\MyExtension\Domain\Model\ScientificInstitution'
```
Comment (mbrodala) [inline Documentation/6-Persistence/5-modeling-the-class-hierarchy.rst:112]:
```suggestion
            'recordType' => '\MyVendor\MyExtension\Domain\Model\ScientificInstitution',
```

File: Documentation/6-Persistence/5-modeling-the-class-hierarchy.rst [hunk 6]
```diff
@@ -82,41 +82,36 @@ The name of the field that contains the type can be chosen freely. In our case,
...
+        ]
```
Comment (mbrodala) [inline Documentation/6-Persistence/5-modeling-the-class-hierarchy.rst:113]:
```suggestion
        ],
```

File: Documentation/6-Persistence/5-modeling-the-class-hierarchy.rst [hunk 7]
```diff
@@ -82,41 +82,36 @@ The name of the field that contains the type can be chosen freely. In our case,
...
+data of the instances are stored and with which type they should be stored. This is done in "Configuration/Extbase/Persistence/Classes.php".
+
+::
+   
+    <?php
+    declare(strict_types = 1);
+
+    return [
+        \MyVendor\MyExtension\Domain\Model\Organization::class => [
+            'tableName' => 'tx_myextension_domain_model_party',
```
Comment (mbrodala) [inline Documentation/6-Persistence/5-modeling-the-class-hierarchy.rst:95]: There are mentions of this afterwards, can you also adjust them?

**Review by mbrodala — CHANGES_REQUESTED** [review]

File: Documentation/6-Persistence/5-modeling-the-class-hierarchy.rst (hunk 7 above)
Comment (jankornblum) [reply Documentation/6-Persistence/5-modeling-the-class-hierarchy.rst:95]: Sorry. Very first time here. What do you mean in detail?

Comment (mbrodala) [reply Documentation/6-Persistence/5-modeling-the-class-hierarchy.rst:95]: I mean this, which refers to the old TypoScript configuration:

image

**Comment by linawolf** [comment]
What is the status of this PR, @mbrodala? Have all required changes been included?

**Review by mbrodala — APPROVED** [review]

**Comment by mbrodala** [comment]
@linawolf Yes, all done.

**Pull request merged**